src/fsharp/NameResolution.fs

```diff
@@ -2199,116 +2197,115 @@ let (|AccessibleEntityRef|_|) amap m ad (modref: ModuleOrNamespaceRef) mspec =
     let eref = modref.NestedTyconRef mspec
     if IsEntityAccessible amap m ad eref then Some eref else None
 
-let rec ResolveExprLongIdentInModuleOrNamespace (ncenv:NameResolver) nenv (typeNameResInfo: TypeNameResolutionInfo) ad resInfo depth m modref (mty:ModuleOrNamespaceType) (lid :Ident list) =
+let rec ResolveExprLongIdentInModuleOrNamespace (ncenv:NameResolver) nenv (typeNameResInfo: TypeNameResolutionInfo) ad resInfo depth m modref (mty:ModuleOrNamespaceType) (id:Ident) (rest :Ident list) =
     // resInfo records the modules or namespaces actually relevant to a resolution
-    match lid with 
-    | [] -> raze(Error(FSComp.SR.nrUnexpectedEmptyLongId(),m))
```

Comment:
I wonder if splitting the id/rest will allow new code paths to be ran that were not before?   Thinking about it logically the is should always be present and what your are doing is removing unrepresentable state.

Reply:
yes we move that decision to outer scope. and don't check over and over again in rec functions